raft/src/main/java/org/apache/kafka/raft/internals/VoterSet.java

```diff
@@ -56,6 +56,18 @@ final public class VoterSet {
         this.voters = voters;
     }
 
+    /**
+     * Returns the node information for all the voters
+     */
+    public Set<Node> voterNodes(ListenerName listenerName) {
+        Set<Node> allVoters = new HashSet<>();
+        for (VoterNode voter : voters.values()) {
+            voter.address(listenerName).ifPresent(address ->
+                allVoters.add(new Node(voter.voterKey().id(), address.getHostString(), address.getPort())));
+        }
+        return allVoters;
+    }
```

Comment:
This should have the same semantic as `voterNodes(Stream, ListenerName)`. Throw an `IllegalArgumentException` is there is a voter that doesn't support the given listener.

Let's add a test for this new functionality.